src/asm/symbol.c

```diff
@@ -390,17 +405,25 @@ struct Symbol *sym_AddEqu(char const *symName, int32_t value)
 struct Symbol *sym_AddString(char const *symName, char const *value)
 {
 	struct Symbol *sym = createNonrelocSymbol(symName);
-	size_t len = strlen(value);
-	char *string = malloc(len + 1);
 
-	if (string == NULL)
-		fatalerror("No memory for string equate: %s\n", strerror(errno));
-	strcpy(string, value);
+	assignStringSymbol(sym, value);
 
-	sym->type = SYM_EQUS;
-	/* TODO: use other fields */
-	sym->macroSize = len;
-	sym->macro = string;
+	return sym;
+}
+
+struct Symbol *sym_RedefString(char const *symName, char const *value)
+{
+	struct Symbol *sym = sym_FindExactSymbol(symName);
+
+	if (!sym) {
+		sym = createsymbol(symName);
+	} else if (sym->type != SYM_EQUS) {
+		error("'%s' already defined as non-EQUS at ", symName);
+		dumpFilename(sym);
+		putc('\n', stderr);
+	}
+
+	assignStringSymbol(sym, value);
```

Comment:
Careful, this leaks the old string.

Reply:
I think `sym_Purge` is leaking strings too. It calls `hash_RemoveElement` and `free(symbol)`, but not `free(symbol->macro)` for `SYM_MACRO` or `SYM_EQUS`.

Reply:
Fixed for `sym_RedefString`.

Reply:
That ought to be fixed too, then; whether in this PR or separately, I'll leave to you.

Reply:
Actually, that causes equs-purge and macro-purge to fail, since they purge themselves.

Reply:
If they are self-referenced? Geh. Please add a `FIXME` comment instead, then...

Reply:
`FIXME` added.

Reply:
This code path doesn't have it, though. Other than that, LGTM: I'll merge once that's done.